Incompatibility between vtk and pathlib for old versions

Hi,

When using the latest code for ASTE on the `develop` branch (commit `9df0e7d0e0d3d4f1e21384531ab68d0ef5e006e9`), I encouter an issue with `precice-aste-join`.

After running this command directly taken from the tutorial:

```
precice-aste-join -m mapped/mapped -o result.vtu --recovery fine_mesh/fine_mesh_recovery.json
```

The following error is raised:

```sh
---[ASTE-Join] INFO : Final mesh contains 16641 points, 82689 cells
Traceback (most recent call last):
  File "/opt/preCICE/aste/bin/precice-aste-join", line 373, in <module>
    MeshJoiner()
  File "/opt/preCICE/aste/bin/precice-aste-join", line 30, in __init__
    self.join(args)
  File "/opt/preCICE/aste/bin/precice-aste-join", line 122, in join
    MeshJoiner.write_mesh(joined_mesh, filename)
  File "/opt/preCICE/aste/bin/precice-aste-join", line 367, in write_mesh
    writer.SetFileName(filename)
TypeError: SetFileName argument 1: string or None required
[ERROR] Command failed: precice-aste-join -m mapped/mapped -o result_f128_s64.vtu --recovery b_mesh/b_mesh_recovery.json

```

As specified, this error can be traced back to the `write_mesh` function in `precice-aste-join`. This error is raised probably because vtk doesn't recognize pathlib objects. **Replacing the line by ` writer.SetFileName(str(filename))` fixes the issue.** This problem is certainly linked to the very old vtk version that I am using (v6.1.0) on an old operating system (CentOS 7). This issue is not present in the ASTE release v3.3.0 (branch `master`).

--- a/changelog-entries/243.md
+++ b/changelog-entries/243.md
@@ -1,0 +1,1 @@
+- Fixed an incompatibility of pathlib with older vtk versions

--- a/src/precice-aste-evaluate
+++ b/src/precice-aste-evaluate
@@ -340,7 +340,7 @@ class Calculator:
             raise ExtensionError(
                 'Unkown input file extension please check your input file or hype "--help" for more information.'
             )
-        reader.SetFileName(in_meshname)
+        reader.SetFileName(str(in_meshname))
         reader.Update()
         vtk_dataset = reader.GetOutput()
         logger.info("Mesh contains {} points.".format(vtk_dataset.GetNumberOfPoints()))
@@ -366,7 +366,7 @@ class Calculator:
             os.makedirs(directory, exist_ok=True)
             out_meshname = os.path.join(directory, out_meshname)
 
-        writer.SetFileName(out_meshname)
+        writer.SetFileName(str(out_meshname))
         writer.Write()
         logger.info(f'Written output to "{out_meshname}".')
 

--- a/src/precice-aste-join
+++ b/src/precice-aste-join
@@ -364,7 +364,7 @@ class MeshJoiner:
             writer = vtk.vtkXMLUnstructuredGridWriter()
         else:
             raise ExtensionError(f"Unknown File extension: {extension}")
-        writer.SetFileName(filename)
+        writer.SetFileName(str(filename))
         writer.SetInputData(meshfile)
         writer.Write()
 

--- a/src/precice-aste-partition
+++ b/src/precice-aste-partition
@@ -605,11 +605,11 @@ class MeshPartitioner:
     @staticmethod
     def vtu2vtk(inmesh, outmesh):
         reader = vtk.vtkXMLUnstructuredGridReader()
-        reader.SetFileName(inmesh)
+        reader.SetFileName(str(inmesh))
         reader.Update()
         mesh = reader.GetOutput()
         writer = vtk.vtkUnstructuredGridWriter()
-        writer.SetFileName(outmesh + ".vtk")
+        writer.SetFileName(str(outmesh) + ".vtk")
         writer.SetInputData(mesh)
         writer.SetFileTypeToBinary()
         writer.Write()

